src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaMessageConsumer.cs

```diff
@@ -117,50 +95,14 @@ void HandleKafkaError(IConsumer<byte[], byte[]> consumer, Error error)
             if (_cancellationTokenSource.IsCancellationRequested)
                 return;
 
-            var activeDispatchCount = _dispatcher.ActiveDispatchCount;
-            if (activeDispatchCount == 0 || error.IsLocalError)
-            {
+            if (error.IsLocalError)
                 _cancellationTokenSource.Cancel();
-                _deliveryComplete.TrySetResult(true);
-            }
-        }
-
-        Task HandleDeliveryComplete()
-        {
-            if (IsStopping)
-            {
-                LogContext.Debug?.Log("Consumer shutdown completed: {InputAddress}", _context.InputAddress);
-
-                _deliveryComplete.TrySetResult(true);
-            }
-
-            return Task.CompletedTask;
-        }
-
-        protected override Task StopAgent(StopContext context)
-        {
-            LogContext.Debug?.Log("Stopping consumer: {InputAddress}", _context.InputAddress);
-
-            SetCompleted(ActiveAndActualAgentsCompleted(context));
-
-            return Completed;
         }
 
-        async Task ActiveAndActualAgentsCompleted(StopContext context)
+        protected override async Task ActiveAndActualAgentsCompleted(StopContext context)
         {
-            if (_dispatcher.ActiveDispatchCount > 0)
-            {
-                try
-                {
-                    await _deliveryComplete.Task.OrCanceled(context.CancellationToken).ConfigureAwait(false);
-                }
-                catch (OperationCanceledException)
-                {
-                    LogContext.Warning?.Log("Stop canceled waiting for message consumers to complete: {InputAddress}", _context.InputAddress);
-                }
-            }
+            await base.ActiveAndActualAgentsCompleted(context).ConfigureAwait(false);
```

Comment:
no stop processing required with Kafka?

Reply:
with kafka you need to `.Close()` when you completed consumption, checkpoint etc. which is done bellow